internal/cache/controller_obj_cache.go

```diff
@@ -2554,6 +2541,51 @@ func validateAndConfigureSeGroup(client *clients.AviClient) bool {
 		return false
 	}
 
+	// Indulge in Marker based SEGroup searching only when user input is empty.
+	// SE Group selection priority
+	// 1. User input
+	seGroupToUse := lib.GetSEGNameEnv()
+
+	// 2. Marker based
+	if seGroupToUse == "" {
+		for _, elem := range elems {
+			seg := models.ServiceEngineGroup{}
+			err = json.Unmarshal(elem, &seg)
+			if err != nil {
+				utils.AviLog.Warnf("Failed to unmarshal data, err: %v", err)
+				continue
+			}
+
+			clusterName := lib.GetClusterName()
+			if lib.GetAdvancedL4() {
+				clusterName = lib.GetClusterID()
+			}
+
+			if len(seg.Markers) == 1 &&
+				*seg.Markers[0].Key == lib.ClusterNameLabelKey &&
+				len(seg.Markers[0].Values) == 1 &&
+				seg.Markers[0].Values[0] == clusterName {
+				seGroupToUse = *seg.Name
+				break
+			}
+		}
+	}
+	// 3. Default-SEGroup
+	if seGroupToUse == "" {
+		seGroupToUse = lib.DEFAULT_SE_GROUP
+	}
+	lib.SetSEGName(seGroupToUse)
+
+	// Not applicable for NodePort mode / disable route is set as True
+	if lib.GetDisableStaticRoute() {
```

Comment:
If you move this downwards, then for NodePort mode, you would still run through the above code logic, which is not necessary.